### PR TITLE
[Examples | Catalog] Updates all components to Swift 4.

### PR DESCRIPTION
**Pull request by codeman7**
Update all components to Swift 4. This covers all Swift examples and Swift files throughout Catalog.

This doesn't close any issues.

### CODE REVIEW

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 1]
```diff
@@ -93,8 +94,8 @@ class MDCNodeListViewController: CBCNodeListViewController {
 
     appBar.navigationBar.tintColor = UIColor.white
     appBar.navigationBar.titleTextAttributes = [
-      NSForegroundColorAttributeName: UIColor.white,
-      NSFontAttributeName: appBarFont ]
+      NSAttributedStringKey.foregroundColor.rawValue: UIColor.white,
```
Comment (willlarche) [inline catalog/MDCCatalog/MDCNodeListViewController.swift:96]: I love this improvement. Just saying.

**Review by willlarche — APPROVED** [review]
Thank you!

**Comment by willlarche** [comment]
Awaiting CI.

**Comment by willlarche** [comment]
This is failing on Travis because it's using Xcode 8.3.3. Let's see if we can update that.

**Comment by willlarche** [comment]
Please update the .travis.yml file's line 2 to be `osx_image: xcode9` in this PR.

**Comment by codeman7** [comment]
Done.

**Comment by willlarche** [comment]
Travis is unhappy with the following lines:
```
/Users/travis/build/material-components/material-components-ios/components/AppBar/examples/AppBarDelegateForwardingExample.swift:35:9: type 'NSAttributedStringKey' (aka 'NSString') has no member 'foregroundColor'
      [ NSAttributedStringKey.foregroundColor.rawValue: UIColor.white ]
        ^~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~
▸ Compiling AppBarImageryExample.swift
❌  /Users/travis/build/material-components/material-components-ios/components/AppBar/examples/AppBarImageryExample.swift:48:9: type 'NSAttributedStringKey' (aka 'NSString') has no member 'foregroundColor'
      [ NSAttributedStringKey.foregroundColor.rawValue: UIColor.white ]
        ^~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~```

**Comment by codeman7** [comment]
Is it because NSAttributedStringKey was added in iOS 11?  Should I add a check to see the OS is iOS 11 or higher or just revert back to the old code?

**Comment by willlarche** [comment]
Ah, yes. We are currently supporting back to iOS 8. Since that's gonna be 2-3 years before we can _exclusively_ support this new enum, let's revert the code if it still works.

**Comment by codeman7** [comment]
I just added some other error fixes that it was throwing in the console. This still shouldn't pass Travis.

**Comment by codeman7** [comment]
So looking at ways to work around NSAttributedStringKey for older OS versions and we could do this but I think it will add a ton of code, I don't know if you think it is worth it. 
screen shot 2017-09-26 at 6 55 34 pm

**Comment by codeman7** [comment]
If Travis passes 

DO NOT APPROVE 

I commented out some lines to hopefully give me more details as to why Travis failed.

**Pull request closed without merging**